redash/query_runner/databend.py

```diff
@@ -0,0 +1,118 @@
+try:
+    from databend_sqlalchemy import connector
+    import re
+
+    enabled = True
+except ImportError:
+    enabled = False
+
+from redash.query_runner import BaseQueryRunner, register
+from redash.query_runner import TYPE_STRING, TYPE_INTEGER, TYPE_BOOLEAN, TYPE_FLOAT, TYPE_DATETIME, TYPE_DATE
+from redash.utils import json_dumps, json_loads
+
+
+class Databend(BaseQueryRunner):
+    noop_query = "SELECT 1"
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "host": {"type": "string", "default": "localhost"},
+                "port": {"type": "int", "default": 8000},
+                "username": {"type": "string", "default": "root"},
+                "password": {"type": "string", "default": ""},
+                "database": {"type": "string"},
+                "secure": {"type": "string", "default": False},
+            },
+            "order": ["username", "password", "host", "port", "database"],
+            "required": ["username", "database"],
+            "secret": ["password"],
+        }
+
+    @classmethod
+    def name(cls):
+        return "Databend"
+
+    @classmethod
+    def type(cls):
+        return "Databend"
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    @staticmethod
+    def _define_column_type(column_type):
+        c = column_type.lower()
+        f = re.search(r"^nullable\((.*)\)$", c)
+        if f is not None:
+            c = f.group(1)
+        if c.startswith("int") or c.startswith("uint"):
+            return TYPE_INTEGER
+        elif c.startswith("float"):
+            return TYPE_FLOAT
+        elif c == "datetime":
+            return TYPE_DATETIME
+        elif c == "date":
+            return TYPE_DATE
+        else:
+            return TYPE_STRING
+
+    def run_query(self, query, user):
+        host = (self.configuration.get("host") or "localhost"),
+        port = (self.configuration.get("port") or 8000),
+        username = (self.configuration.get("username") or None),
+        password = (self.configuration.get("password") or None),
+        database = (self.configuration.get("database") or None),
+        secure = (self.configuration.get("secure") or False),
+        connection = connector.connect(f"databend://{username}:{password}@{host}:{port}/{database}?secure={secure}")
```

Comment:
This might break if one of the fields has special characters (`:` for example). You need to use something like `urlencode` to properly escape the values first.

Reply:
I use the `urlparse(url)` in databend sqlalchemy and it can parse the correct param even the fields has special characters (: for example). 
<img width="776" alt="image" src="https://user-images.githubusercontent.com/7600925/226507847-83d73b3c-91d0-4d44-8e5b-2ed798f04637.png">

Reply:
Interesting.. we had this issue in the past, but maybe it wasn't using urlparse. 👍